x-pack/plugins/security/public/lib/api.ts

```diff
@@ -5,45 +5,52 @@
  */
 
 import { kfetch } from 'ui/kfetch';
-import { AuthenticatedUser, Role, User } from '../../common/model';
+import { AuthenticatedUser, Role, User, EditUser } from '../../common/model';
 
 const usersUrl = '/api/security/v1/users';
 const rolesUrl = '/api/security/role';
 
 export class UserAPIClient {
-  public static async getCurrentUser(): Promise<AuthenticatedUser> {
+  public async getCurrentUser(): Promise<AuthenticatedUser> {
     return await kfetch({ pathname: `/api/security/v1/me` });
   }
 
-  public static async getUsers(): Promise<User[]> {
+  public async getUsers(): Promise<User[]> {
     return await kfetch({ pathname: usersUrl });
   }
 
-  public static async getUser(username: string): Promise<User> {
+  public async getUser(username: string): Promise<User> {
     const url = `${usersUrl}/${encodeURIComponent(username)}`;
     return await kfetch({ pathname: url });
   }
 
-  public static async deleteUser(username: string) {
+  public async deleteUser(username: string) {
     const url = `${usersUrl}/${encodeURIComponent(username)}`;
     await kfetch({ pathname: url, method: 'DELETE' }, {});
   }
 
-  public static async saveUser(user: User) {
+  public async saveUser(user: EditUser, isNewUser: boolean) {
     const url = `${usersUrl}/${encodeURIComponent(user.username)}`;
-    await kfetch({ pathname: url, body: JSON.stringify(user), method: 'POST' });
+    const saveUserData: Record<string, any> = {
+      ...user,
+    };
+    if (!isNewUser) {
```

Comment:
> What if we were to introduce a CreateUser model which extends from EditUser and specifies password as a field? 

`CreateUser` would end up extending from `User` at this point, because `EditUser` would disappear (the only fields it defines are `password` and `confirmPassword`).

> And then we could exclude confirmPassword from both of these two models, as they're primarily concerns of the actual UI itself?

Primarily yes, but this would end up complicating the `UserValidator` function signatures, as `confirmPassword` would need to be passed in separately into both `validateForSave`, which is called in a handful of places.

> nit: it feels rather awkward having to delete the confirmPassword and conditionally the password here.

I do agree that it's awkward to do this here. Would you be opposed to meeting this halfway by keeping the interface as-is, but move the translation to the `EditUserPageUI` where we do the role transform?

Reply:
> I do agree that it's awkward to do this here. Would you be opposed to meeting this halfway by keeping the interface as-is, but move the translation to the EditUserPageUI where we do the role transform?

Sure, I'm fine with that.